packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts

```diff
@@ -0,0 +1,252 @@
+import { Web3Wrapper } from '@0x/dev-utils';
+import { TakerRequestQueryParams, V4RFQFirmQuote, V4RFQIndicativeQuote } from '@0x/quote-server';
+import { BigNumber, logUtils } from '@0x/utils';
+import { AxiosInstance } from 'axios';
+
+import {
+    AltFirmQuoteResponse,
+    AltIndicativeQuoteReponse,
+    AltOffering,
+    AltQuoteModel,
+    AltQuoteRequestData,
+    AltQuoteSide,
+    AltRfqtMakerAssetOfferings,
+} from '../types';
+
+const IMPUTED_EXPIRY_SECONDS = 120;
+
+function getAltMarketInfo(
+    offerings: AltOffering[],
+    buyTokenAddress: string,
+    sellTokenAddress: string,
+): AltOffering | undefined {
+    for (const offering of offerings) {
+        if (
```

Comment:
Alternative implementation suggestion starting L24 (inside loop), just something to think about
```
for (const offering of offerings) {
    const requestSignature = [buyTokenAddress, sellTokenAddress]
        .sort()
        .join('-')
        .toLowerCase();

    const offeringSignature = [offering.baseAsset, offering.quoteAsset]
        .sort()
        .join('-')
        .toLowerCase();
    if (offeringSignature === requestSignature) {
        return offering
    }
```